broken_link_checker/checker.py

```diff
@@ -103,10 +110,11 @@ def check(self, url: str) -> urllib3.response.HTTPResponse | None:
             return
 
         # We verify the response status
-        if response.status == 200:
-            return response
+        # 2xx stand for request was successfully completed
+        if int(response.status / 100) == 2:
+            return response if self.conn.is_same_host(url) else None
```

Comment:
a `301` is also a code that can stand for a 'valid link', cuz it's a redirection status code !

As i suggested on your other Pull Request, 
i prefer a versatile check (in)!

```suggestion
        if response.status in VALID_STATUS_CODE
            return response if self.conn.is_same_host(url) else None
```
With `VALID_STATUS_CODE` a list of all 2xx, 3xx valid status codes standing for a link that is alive.


---------

With your current implementation, if am having in my website a link that is redirecting to another website (and returning 301 as status code), the checker will tell me the link is broken... and that's not true... i don't know if you get what am saying @pythonbrad ?

Reply:
ok for 3xx, the auto redirect is enable for all request. Then if a link A (301) redirect to B (200), in the end we will have B (200). The link will  be considered broken just if we have too much redirection.

Reply:
ahhhh okkkk i see, it's all good then

Reply:
Good!

Reply:
> 

Just to clarify something. Generally a http status code 2xx stand for success.
Between just verify that we have a 2xx status code or verify that the status got in inside the VALID_STATUS_CODE, what do you propose?

```python
VALID_STATUS_CODE=(200,` 201, 202, 203, 204, 205, 206, 207, 208, 226)
```

Reply:
> With `VALID_STATUS_CODE` a list of all 2xx, 3xx valid status codes standing for a link that is alive

@Sanix-Darker  Just to clarify something. Generally a http status code 2xx stand for success.
Between just verify that we have a 2xx status code or verify that the status got in inside the VALID_STATUS_CODE, what do you propose?

```python
VALID_STATUS_CODE=(200,` 201, 202, 203, 204, 205, 206, 207, 208, 226)
```

Reply:
> With `VALID_STATUS_CODE` a list of all 2xx, 3xx valid status codes standing for a link that is alive.

@Sanix-Darker 
Just to clarify something. Generally a http status code 2xx stand for success.
Between just verify that we have a 2xx status code or verify that the status got in inside the VALID_STATUS_CODE, what do you propose?

```python
VALID_STATUS_CODE=(200,` 201, 202, 203, 204, 205, 206, 207, 208, 226)
```

Reply:
sound good for me @pythonbrad